### PR TITLE
Change non-legacy control points to have CustomSampleBank 1

### LINKED ISSUE #29028: Custom hitsounds do not persist using "Edit externally"
### Type

Game behaviour

### Bug description

The "Edit externally" feature does work for adding custom hitsounds, but they do not persist in the editor after save > exit > reload, and do not persist during gameplay.

### Screenshots or videos

[2024-07-23 12-48-56.webm](https://github.com/user-attachments/assets/6ff925cb-9057-46f4-b07f-7f4cf5f1a5d5)

### Version

2024.718.1-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16351826/compressed-logs.zip)

### PR DESCRIPTION
**Pull request by kstefanowicz**
- Closes #29028

Passes test `osu.Game.Tests.Beatmaps.Formats.LegacyBeatmapEncoderTest`, which previous implementation #29084 failed (thanks @smoogipoo!)

See also #29280 regarding long-term implementation of custom samplesets - this PR implements the third option proposed.

### CODE REVIEW

**Comment by smoogipoo** [comment]
Maybe @OliBomby or @peppy can provide feedback on whether this is the direction we want to go? From my reading of it this means any lazer-added sample bank will be CSS1 i.e. it will be overridden by whatever hitsample file exists in the beatmap.

**Comment by OliBomby** [comment]
I do agree with the direction of allowing sample banks to be overriden this way. There is no need for custum sample set indices because you can use custom bank names instead (though these cant be saved yet). 

However this implementation seems to still be somewhat broken. When I add a custom sample, I still need to cut and repaste the objects for the custom sound to play, and when I reload the editor or go in play the sounds are gone again.

**Comment by kstefanowicz** [comment]
> However this implementation seems to still be somewhat broken. When I add a custom sample, I still need to cut and repaste the objects for the custom sound to play, and when I reload the editor or go in play the sounds are gone again.

Are you testing on an existing beatmap? This change only affects creating new timing points, it will not change existing ones that are already saved as 0.

**Comment by OliBomby** [comment]
> Are you testing on an existing beatmap? This change only affects creating new timing points, it will not change existing ones that are already saved as 0.

I'm testing on a completely new beatmap. It's still very much broken.

https://github.com/user-attachments/assets/ef2c3d25-83bb-46b2-bbd8-9e45e63bcc09


https://github.com/user-attachments/assets/c3ae2ac6-9705-45c9-93f6-c5e510e78a55

**Comment by kstefanowicz** [comment]
I have changed this logic in `getLegacyControlPointProperties` to create `LegacyHitSampleInfo` with customSampleBank 1 if the sample point isn't already legacy. Still passes `LegacyBeatmapEncoderTest`.

This appears to work at first, but after placing a hit object the timing point reverts back to 0:

[pr-29311-1.webm](https://github.com/user-attachments/assets/f6a305b0-5030-4fae-8a9c-9f1fcc928840)

Looking at debug, after the encoder is called as part of EditorChangeHandler, samplePoint is being passed with CustomSampleBank -1:
image

I was not able to determine why this is - I will continue investigating

**Comment by bdach** [comment]
Before further review happens here I will link back https://github.com/ppy/osu/discussions/29280 here, and particularly https://github.com/ppy/osu/discussions/29280#discussioncomment-10506039 which seems to indicate that this implementation is much more opinionated about what should happen than it probably should be.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs [hunk 1]
```diff
@@ -220,8 +220,15 @@ LegacyControlPointProperties getLegacyControlPointProperties(ControlPointGroup g
                 var samplePoint = legacyControlPoints.SamplePointAt(group.Time);
                 var effectPoint = legacyControlPoints.EffectPointAt(group.Time);
 
+                // if samplePoint isn't already legacy, create LegacyHitSampleInfo with customSampleBank 1
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs:223]: If we're going to go with this, I want much better inline documentation describing what the expectations are, and what `1` means.

**Comment by OliBomby** [comment]
As pointed out by @bdach and https://github.com/ppy/osu/discussions/29280#discussioncomment-10506039 there are some fundamental issues with this implementation that I failed to notice earlier, and I think we should go in a different direction.

**Comment by peppy** [comment]
Going to close this for now based on https://github.com/ppy/osu/pull/29311#issuecomment-2385563246 and the fact that it's only a few lines of change.

**Pull request closed without merging**